File: frontend/src/types/provider-status.ts
export type ProviderHealthStatus = "ok" | "error" | "not_configured";

export type ProviderStatusItem = {
  provider: string;
  status: ProviderHealthStatus;
  message: string;
  checked_at: string;
};

export type ProviderStatusResponse = {
  items: ProviderStatusItem[];
};

File: backend/app/schemas/provider_status.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


ProviderHealthStatus = Literal["ok", "error", "not_configured"]


class ProviderStatusItem(BaseModel):
    provider: str
    status: ProviderHealthStatus
    message: str
    checked_at: datetime


class ProviderStatusResponse(BaseModel):
    items: list[ProviderStatusItem]
